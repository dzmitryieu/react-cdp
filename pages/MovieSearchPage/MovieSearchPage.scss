@import "../../src/utils/constants.scss";

.wrapper {
  background-image: url(../../src/assets/images/cover-image.jpg);
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center center;
}

.section {
  padding: 0 60px 50px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 600px;
  position: relative;
  background: rgba(0, 0, 0, 0.65);
}

.img {
  position: absolute;
  top: 0;
  left: 0;
  width: 100vw;
}

@media screen and (max-width: $tablet-size-screen) {
  .section {
    padding: 0 20px;
  }
}
