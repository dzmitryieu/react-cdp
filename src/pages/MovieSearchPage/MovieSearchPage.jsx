import React from 'react';
import PropTypes from 'prop-types';
import styles from './MovieSearchPage.scss';
import Header from '../../components/Header';
import Search from '../../components/Search';
import '../../assets/images/cover-image.jpg';

const MovieSearchPage = ({ isShowSearchPage }) => {
  if (!isShowSearchPage) return null;
  return (
    <div id="MovieSearchPage" className={styles.wrapper}>
      <section className={styles.section}>
        <Header />
        <Search />
      </section>
    </div>
  );
};

MovieSearchPage.propTypes = {
  isShowSearchPage: PropTypes.bool.isRequired,
};

export default MovieSearchPage;
