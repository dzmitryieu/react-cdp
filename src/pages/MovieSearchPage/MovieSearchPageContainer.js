import { connect } from 'react-redux';
import MovieSearchPage from './MovieSearchPage';
import { isShowSearchPage } from '../../modules/film/filmSelectors';


const mapStateToProps = state => ({
  isShowSearchPage: isShowSearchPage(state),
});

export default connect(mapStateToProps)(MovieSearchPage);
