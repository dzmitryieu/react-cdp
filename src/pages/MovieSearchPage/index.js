import MovieSearchPage from './MovieSearchPageContainer';

export default MovieSearchPage;
