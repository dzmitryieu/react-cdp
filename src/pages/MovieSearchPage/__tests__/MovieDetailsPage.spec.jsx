import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import MovieSearchPage from '../index';

const shallow = new ShallowRenderer();

describe('MovieSearchPage Snapshot', () => {
  test('renders', () => {
    const component = shallow.render(
      <MovieSearchPage />,
    );
    expect(component).toMatchSnapshot();
  });
  test('renders isActive', () => {
    const component = shallow.render(
      <MovieSearchPage isActive />,
    );
    expect(component).toMatchSnapshot();
  });
});
