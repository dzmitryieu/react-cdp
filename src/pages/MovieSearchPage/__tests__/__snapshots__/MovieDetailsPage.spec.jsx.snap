// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieSearchPage Snapshot renders 1`] = `
<div
  className="wrapper"
  id="MovieSearchPage"
>
  <section
    className="section"
  >
    <Header />
    <Search />
  </section>
</div>
`;

exports[`MovieSearchPage Snapshot renders isActive 1`] = `
<div
  className="wrapper"
  id="MovieSearchPage"
>
  <section
    className="section"
  >
    <Header />
    <Search />
  </section>
</div>
`;
